core/src/main/kotlin/org/evomaster/core/problem/rest/service/AbstractRestSampler.kt

```diff
@@ -81,18 +83,24 @@ abstract class AbstractRestSampler : HttpWsSampler<RestIndividual>() {
         val openApiURL = problem.openApiUrl
         val openApiSchema = problem.openApiSchema
 
+        // set up authentications moved up since we are going to get authentication info from HttpWsSampler
+        setupAuthentication(infoDto)
+
         if(!openApiURL.isNullOrBlank()) {
-            swagger = OpenApiAccess.getOpenAPIFromURL(openApiURL)
+
+            try {
+                retrieveSwagger(openApiURL)
+            }
+            catch (e : Exception) {
+                e.printStackTrace()
```

Comment:
printing should always go through a log object. but anyway this catch needs to be removed

Reply:
Hi Andrea,

Thanks for your comment. I have already removed try/catch block and just kept the method retrieveSwagger, so this comment has also been applied.